.github/workflows/pull-request.yml

```diff
@@ -0,0 +1,34 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: pull request
+
+on:
+  pull_request:
+    types:
+      - reopened
+      - opened
+      - edited
+      - synchronize
+
+jobs:
+  title:
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Check pull request title
+        env:
+          TITLE: ${{ github.event.pull_request.title }}
+        run:
+          echo "${TITLE}" | grep -E '^HDDS-[1-9][0-9]{3,4}\. .*[^ ]$' | grep -v '  '
```

Comment:
If you want to ensure there is not a double space after the `HDDS-1234.` should we add in something explicit, like:

```
^HDDS-[1-9][0-9]{3,4}\.[[:space:]][^[:space:]]+.*[^[:space:]]$
```

Ie, it is a space; zero or 1 "not a space"; zero or more of any character.

Reply:
 The current expression is checking no double space at any place, and I think it's better (and clearer).

Reply:
Ah ok. I didn't realize the check was for double spaces everywhere. Makes sense to keep it as it is then.